File: Harjoitus_1/exercise1_3.py
def main():
    keskikulutus = float(6.5)
    try:
        matkaInKm = float(input("Anna matka kilometreinä: "))
    except ValueError:
        print("Virheellinen syöte. Anna numero.")
        return 0
    if matkaInKm < 0:
        print("Matka ei voi olla negatiivinen")
        return 0
    kokonaiskulutus = (keskikulutus * matkaInKm) / 100
    print(f"Polttoaineen kulutus on {kokonaiskulutus:.1f} litraa")

if __name__ == "__main__":
    main()

File: cpp/Harjoitus_1/exercise1_3.cpp
#include <iostream>
#include <limits>
#include <iomanip>

float distance;
float economy = 6.5;

int main() {
    std::cout << "Anna matka:\n> ";
    while (!(std::cin >> distance)){
        std::cout << "Väärä syöte. Anna luku:\n> ";
        std::cin.clear();
        std::cin.ignore(std::numeric_limits<std::streamsize>::max(),'\n');
    }
    if (distance <= 0) {
        std::cout << "Matka ei voi olla yhtäkuin tai alle 0";
        distance = 0;
        main();
    }
    std::cout << "Polttoaineen kulutus on: " << economy * distance / 100 << " litraa\n\n";
    return 0;
}